test/component/uk/gov/hmrc/individualsincomeapi/controllers/v2/LiveSaIncomeControllerSpec.scala

```diff
@@ -0,0 +1,568 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package component.uk.gov.hmrc.individualsincomeapi.controllers.v2
+
+import java.util.UUID
+
+import component.uk.gov.hmrc.individualsincomeapi.stubs.{AuthStub, BaseSpec, DesStub, IndividualsMatchingApiStub}
+import org.joda.time.LocalDate
+import play.api.libs.json.Json
+import play.api.test.Helpers._
+import scalaj.http.Http
+import uk.gov.hmrc.domain.{Nino, SaUtr}
+import uk.gov.hmrc.individualsincomeapi.domain.{DesSAIncome, DesSAReturn, SAIncome, TaxYear}
+
+class LiveSaIncomeControllerSpec extends BaseSpec {
+
+  val matchId = UUID.randomUUID().toString
+  val nino = Nino("AA100009B")
+  val fromTaxYear = TaxYear("2016-17")
+  val toTaxYear = TaxYear("2018-19")
+  val desIncomes = Seq(
+    DesSAIncome(
```

Comment:
V2 controller integrates with IF not DES. Can we remove the references to DES as they should not be needed for not yet implemented?